nixos/modules/services/system/nohang.nix

```diff
@@ -0,0 +1,105 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+let
+  cfg = config.services.nohang;
+
+  inherit (lib)
+    literalExpression
+    mkEnableOption
+    mkIf
+    mkOption
+    mkPackageOption
+    types
+    ;
+in
+{
+  meta = {
+    maintainers = with lib.maintainers; [ Dev380 ];
+  };
+
+  options.services.nohang = {
+    enable = mkEnableOption "nohang, a daemon that keeps system responsiveness when Linux is out of memory";
+
+    package = mkPackageOption pkgs "nohang" { };
+
+    config = mkOption {
+      type = types.either (types.enum [
+        "basic"
+        "desktop"
+      ]) types.path;
+      default = "desktop";
+      example = literalExpression "./my-nohang-config.conf";
+      description = ''
+        Configuration file to use with nohang. The default and desktop example configurations in the nohang repository
+        can be used by setting this to "basic" or "desktop" (which is the default). Otherwise, you can set it to the path
+        of a custom configuration file.
+      '';
+    };
+  };
+
+  config = mkIf cfg.enable {
+    systemd.services.nohang = {
+      description = "Sophisticated low memory handler";
+      documentation = [
+        "man:nohang(8)"
+        "https://github.com/hakavlad/nohang"
+      ];
+      after = [ "sysinit.target" ];
+      wantedBy = [ "multi-user.target" ];
+      serviceConfig = {
+        ExecStart =
+          "${cfg.package}/bin/nohang --monitor --config "
+          + (
+            if cfg.config == "basic" then
+              "${cfg.package}/etc/nohang/nohang.conf"
+            else if cfg.config == "desktop" then
+              "${cfg.package}/etc/nohang/nohang-desktop.conf"
+            else
+              cfg.config
+          );
+        Slice = "hostcritical.slice";
+        SyslogIdentifier =
+          if cfg.config == "basic" then
+            "nohang"
+          else if cfg.config == "desktop" then
+            "nohang-desktop"
+          else
+            "nohang-custom-config";
```

Comment:
Is this something that is done upstream?

Reply:
Yes I just copied the 2 default configs from upstream: [nohang-desktop.service](https://github.com/hakavlad/nohang/blob/master/systemd/nohang-desktop.service.in) and [nohang.service](https://github.com/hakavlad/nohang/blob/master/systemd/nohang.service.in), and the README says
> nohang-desktop.conf provides default settings optimized for desktop usage.

so I assumed that's meant to be the default